pkg/controller/webhook/federatedtypeconfig_webhook.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"encoding/json"
+	"net/http"
+	"sync"
+
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
+	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1/validation"
+)
+
+type FederatedTypeConfigValidationHook struct {
+	client dynamic.ResourceInterface
+
+	lock        sync.RWMutex
+	initialized bool
+}
+
+func (a *FederatedTypeConfigValidationHook) ValidatingResource() (plural schema.GroupVersionResource, singular string) {
+	return schema.GroupVersionResource{
+			Group:    "admission.core.federation.k8s.io",
+			Version:  "v1alpha1",
+			Resource: "federatedtypeconfigs",
+		},
+		"federatedtypeconfig"
+}
+
+func (a *FederatedTypeConfigValidationHook) Validate(admissionSpec *admissionv1beta1.AdmissionRequest) *admissionv1beta1.AdmissionResponse {
```

Comment:
The code here and that for `FederatedClusterValidationHook` looks quite similar. Recommend to move it out to a common code which can take respective `object` (`admittingObject` in the code) and gvk strings as arguments.

Reply:
> I am assuming atleast this portion of code for rest of the types needing validation will also be same.

Can we do that once we have things fully implemented for other types so that we can avoid pre-factoring?

Reply:
Sure.